module/dkg/client.go

```diff
@@ -0,0 +1,218 @@
+	}
+
+	// TODO: convert from `cadence.Array` to `[]messages.DKGMessage`
+	return dkgMessages.ToGoValue().([]messages.DKGMessage), nil
+}
+
+// SubmitResult submits the final public result of the DKG protocol. This
+// represents the group public key and the node's local computation of the
+// public keys for each DKG participant
+func (c *Client) SubmitResult(groupPublicKey crypto.PublicKey, publicKeys []crypto.PublicKey) error {
+
+	ctx := context.Background()
+
+	// get account for given address
+	account, err := c.flowClient.GetAccount(ctx, c.accountAddress, grpc.EmptyCallOption{})
+	if err != nil {
+		return fmt.Errorf("could not get account details: %v", err)
+	}
+
+	// get latest sealed block to execute transaction
+	latestBlock, err := c.flowClient.GetLatestBlock(ctx, true)
+	if err != nil {
+		return fmt.Errorf("could not get latest block from node: %w", err)
+	}
+
+	tx := sdk.NewTransaction().
+		SetScript(templates.GenerateSendDKGFinalSubmissionScript(c.env)).
+		SetGasLimit(9999).
+		SetReferenceBlockID(latestBlock.ID).
+		SetProposalKey(c.accountAddress, int(c.accountKeyIndex), account.Keys[int(c.accountKeyIndex)].SequenceNumber).
+		SetPayer(c.accountAddress).
+		AddAuthorizer(c.accountAddress)
+
+	// Note: We need to make sure that we pull the keys out in the same order that
+	// we have done here. Group Public key first followed by the individual public keys
+	finalSubmission := make([]cadence.Value, len(publicKeys))
+	finalSubmission = append(finalSubmission, cadence.NewString(groupPublicKey.String()))
+	for _, publicKey := range publicKeys {
+		finalSubmission = append(finalSubmission, cadence.NewString(publicKey.String()))
+	}
+
+	err = tx.AddArgument(cadence.NewArray(finalSubmission))
+	if err != nil {
+		return fmt.Errorf("could not add argument to transaction: %v", err)
+	}
+
+	// sign payload using account signer
+	err = tx.SignPayload(c.accountAddress, int(c.accountKeyIndex), c.signer)
+	if err != nil {
+		return fmt.Errorf("could not sign transaction: %w", err)
+	}
+
+	// submit signed transaction to node
+	txID, err := c.submitTx(tx)
+	if err != nil {
+		return fmt.Errorf("failed to submit transaction: %w", err)
+	}
+
+	// wait for transaction to be sealed
+	result := &sdk.TransactionResult{Status: sdk.TransactionStatusUnknown}
+	for result.Status != sdk.TransactionStatusSealed {
+		result, err = c.flowClient.GetTransactionResult(ctx, txID)
+		if err != nil {
+			return fmt.Errorf("could not get transaction result: %w", err)
+		}
+
+		// if the transaction has expired we skip waiting for seal
+		if result.Status == sdk.TransactionStatusExpired {
+			return fmt.Errorf("submit final dkg result transaction has expired")
+		}
+
+		// wait 1 second before trying again.
+		time.Sleep(time.Second)
+	}
+
+	return nil
+}
+
+// submitTx submits a transaction to the flow network and checks that the
+// transaction is signed
+func (c *Client) submitTx(tx *sdk.Transaction) (sdk.Identifier, error) {
+	ctx := context.Background()
```

Comment:
```suggestion
func (c *Client) submitTx(ctx context.Context, tx *sdk.Transaction) (sdk.Identifier, error) {
```

We should propagate the same context from the parent scope. The main public `DKGClient` methods should actually probably accept a context as argument from their parent scope as well -- that's my bad not including that in the interface.